### ISSUE TITLE
Update CPU usage algorithm

### ISSUE THREAD

**Issue by overvenus** [tikv/tikv#5008]
For now, TiKV uses a simple algorithm to calculate CPU usage, but it overlooks some metrics, eg, irq, softirq, steal. This leads to inaccurate CPU usage, lower than the actual. 

https://github.com/tikv/tikv/blob/c0359b4f0777c930d12551587a60d8a46cc0860e/components/tikv_util/src/metrics/threads_linux.rs#L273-L275

Based on the answer from Vangelis Tasoulas [`Accurate calculation of CPU usage given in percentage in Linux?`](https://stackoverflow.com/questions/23367857/accurate-calculation-of-cpu-usage-given-in-percentage-in-linux/23376195#23376195), we can calculate the CPU usage percentage like:

```
PrevIdle = previdle + previowait
Idle = idle + iowait

PrevNonIdle = prevuser + prevnice + prevsystem + previrq + prevsoftirq + prevsteal
NonIdle = user + nice + system + irq + softirq + steal

PrevTotal = PrevIdle + PrevNonIdle
Total = Idle + NonIdle

# differentiate: actual value minus the previous one
totald = Total - PrevTotal
idled = Idle - PrevIdle

CPU_Percentage = (totald - idled)/totald
```

**Comment by psinghal20** [comment]
@overvenus I think this approach is to calculate the % usage of a cpu for the machine and is not specific to a particular process whereas I think what we need is to do is find %CPU usage for a particular process. 

If that is correct, we can maybe utilise the approach used by [htop](https://github.com/hishamhm/htop). They use `/proc/stat` to get the total period of cpu time and `proc/[pid]/stat` to get the total user and system time for the process. 
- Calculate the total time spent in user and system mode for the process
- Get the previous total time spent in user and system mode for the process
- Calculate the total time spent for the cpu which can be read from `proc/stat`
- Cpu % = ( (NewTotal - PrevTotal) / timePeriod for cpu ) * 100

Reference from htop code:
- CPU usage calculation: https://github.com/hishamhm/htop/blob/master/linux/LinuxProcessList.c#L832-L834
- CPU time period calculation from `/proc/stat`: https://github.com/hishamhm/htop/blob/master/linux/LinuxProcessList.c#L967-L1035

**Comment by overvenus** [comment]
@psinghal20 You are right, also we can apply your formula to `/proc/[pid]/task/[tid]/stat` to calculate CPU usage for an individual thread.

**Comment by jadireddi** [comment]
@overvenus @psinghal20  Currently in `tikv_util` package, https://github.com/tikv/procinfo-rs  `/proc/<pid>/stat` interface is being used  to get the total user and system time of the process . procinfo-rs repo currently doesn't support `proc/stat` interface for getting total time of the cpu. 

As part of this PR, do we need to add `proc/stat` interface  in `procinfo-rs`  repo and later use it in `metrics/threads_linux.rs` for calculating `Cpu %`

**Comment by aknuds1** [comment]
I'm totally new to TiKV, but I could try to solve this :)

**Comment by psinghal20** [comment]
@jadireddi Yes, I think it would be best to add that functionality in the `procinfo-rs`.
CC @overvenus.
